### PR TITLE
Clarify internal (state) API, add custom error types

### PR DESCRIPTION
**Pull request by zer0main**

### CODE REVIEW

**Comment by zer0main** [comment]
Check out `state/api.go` for current state's API overview.

File: pkg/state/api.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		return 0
+	}
+}
+
+// State represents overall Node's state.
+// Data retrievals (e.g. account balances), as well as modifiers (like adding or rolling back blocks)
+// should all be made using this interface.
+type State interface {
+	// Block getters.
+	GetBlock(blockID crypto.Signature) (*proto.Block, error)
```
Comment (Frozen) [inline pkg/state/api.go]: remove "Get" prefix from methods
https://golang.org/doc/effective_go.html#Getters

File: pkg/state/accountsstorage.go [hunk 2]
```diff
@@ -128,14 +128,14 @@ func (s *localStor) reset() {
 	s.assets = make(map[assetBalanceKey][]byte)
 }
 
-type ID2Height interface {
-	HeightByBlockID(blockID crypto.Signature) (uint64, error)
+type iD2Height interface {
```
Comment (alexeykiselev) [inline pkg/state/accountsstorage.go]: May be `idToHeight` better?

File: pkg/state/api.go (hunk 1 above)
Comment (zer0main) [reply pkg/state/api.go]: Done.

File: pkg/state/accountsstorage.go (hunk 2 above)
Comment (zer0main) [reply pkg/state/accountsstorage.go]: Done.

**Pull request merged**